src/app.module.ts

```diff
@@ -1,13 +1,27 @@
-/* eslint-disable prettier/prettier */
 import { Module } from '@nestjs/common';
-//import { AppController } from './app.controller';
-//import { AppService } from './app.service';
-import { CatModule } from './cat/cat.module';
+import { ConfigModule, ConfigService } from '@nestjs/config'
+import { MongooseModule } from '@nestjs/mongoose';
+import { AppService } from './app.service';
+import { AppController } from './app.controller';
+import { ProductModule } from './product/product.module';
 
 @Module({
   imports: [
-    CatModule,
+    ConfigModule.forRoot({
+      isGlobal: true,
+      envFilePath: '.env',
+    }),
+    MongooseModule.forRootAsync({
+      imports: [ConfigModule],
+      inject: [ConfigService],
+      useFactory: async (config: ConfigService) => ({
+        uri: config.get<string>('MONGODB_URI'),
+      })
+    }),
```

Comment:
Im so confuse with this line compare with the database config, ask you, what are you really want to use "uri" or full config like things contain in your db.config ? @Mrpk2109

Reply:
FYI: you can connect to db by many options like use the uri => means completely data contains in 1 line 

and use the config object which is require "dbname username password ur"